macos/FluentUI/Button/Button.swift

```diff
@@ -175,304 +271,355 @@ open class Button: NSButton {
-				x = titleSize.width + Button.titleAndImageRectInterspacing + Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageBelow:
-				x = (rect.width - imageSize.width) / 2
-				y = Button.verticalEdgePadding + titleSize.height + Button.titleAndImageRectInterspacing
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageAbove:
-				x = (rect.width - imageSize.width) / 2
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageOverlaps:
-				x = Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = rect.width - (Button.horizontalEdgePadding * 2)
-				height = rect.height - (Button.verticalEdgePadding * 2)
-				break;
-			case .imageLeading:
-				x = isRTL ? titleSize.width + Button.titleAndImageRectInterspacing + Button.horizontalEdgePadding : Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
-				break
-			case .imageTrailing:
-				x = isRTL ? Button.horizontalEdgePadding : titleSize.width + Button.titleAndImageRectInterspacing + Button.horizontalEdgePadding
-				y = Button.verticalEdgePadding
-				width = imageSize.width
-				height = imageSize.height
+	/// This color is used for the background in the primary style, the pressed background in the secondary
+	/// style, and the content tint (i.e. foreground text/image) color in the borderless style.  It is not used
+	/// for the acrylic style.
+	@objc public var accentColor: NSColor? = Colors.primary
+
+	/// Any of several pre-set button styles.  Setting this property to `.none` does nothing.  Setting it to
+	/// any other value determines the content tint (i.e. foreground text/image), background and border
+	/// color properties for all button states, discarding any previous values.
+	@objc open var style: ButtonStyle = .primary {
+		willSet {
+			switch newValue {
+			case .primary:
+				contentTintColor = ButtonColor.neutralInverted
+				contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+				backgroundColor = accentColor
+				backgroundColorDisabled = ButtonColor.brandBackgroundDisabled
+				borderColor = .clear
+			case .secondary:
+				contentTintColor = .textColor
+				contentTintColorPressed = ButtonColor.neutralInverted?.withSystemEffect(.pressed)
+				backgroundColor = ButtonColor.neutralBackground2
+				backgroundColorPressed = accentColor?.withSystemEffect(.pressed)
+				borderColor = ButtonColor.neutralStroke2
+				borderColorPressed = .clear
+			case .acrylic:
+				contentTintColor = ButtonColor.neutralForeground3
+				backgroundColor = ButtonColor.neutralBackground3
+				borderColor = .clear
+			case .borderless:
+				contentTintColor = accentColor
+				contentTintColorDisabled = ButtonColor.brandForegroundDisabled
+				backgroundColor = .clear
+				borderColor = .clear
+			case .none:
```

Comment:
None of these have breaks at the end, I think that causes some problems with fallthrough?

Reply:
There is no automatic fallthrough in swift switches.  You need to explicitly use the 'fallthrough' keyword for that to happen.  The 'break' keyword in the empty cases at the bottom of the switch may be misleading: in that situation, it serves the same purpose as a 'pass' in python, in that its only function is to satisfy the compiler, which requires a statement to be there.

Reply:
TIL :D